qemu/tests/qemu_guest_agent.py

```diff
@@ -390,6 +393,250 @@ def gagent_check_set_vcpus(self, test, params, env):
+        mount_disk_cmd = mount_disk_cmd.replace("DISK", device_name)
+        session.cmd(mount_disk_cmd)
+
+        error.context("write the disk with dd command", logging.info)
+        write_disk_cmd = params["write_disk_cmd"]
+        session.cmd(write_disk_cmd)
+
+        error.context("Delete the file created before on disk", logging.info)
+        delete_file_cmd = params["delete_file_cmd"]
+        session.cmd(delete_file_cmd)
+
+        # check the bitmap before trim
+        bitmap_before_trim = get_allocation_bitmap()
+        if not re.match(r"\d+-\d+", bitmap_before_trim):
+            raise error.TestFail("didn't get the bitmap of the target disk")
+        error.context("the bitmap_before_trim is %s" % bitmap_before_trim,
+                      logging.info)
+        total_block_before_trim = abs(sum([eval(i) for i in
+                                      bitmap_before_trim.split(',')]))
+        error.context("the total_block_before_trim is %d"
+                      % total_block_before_trim, logging.info)
+
+        error.context("execute the guest-fstrim cmd", logging.info)
+        ret = self.gagent.fstrim()
+        if ret is False:
+            raise error.TestNAError("the guest-fstrim cmd is not supported")
+
+        # check the bitmap after trim
+        bitmap_after_trim = get_allocation_bitmap()
+        if not re.match(r"\d+-\d+", bitmap_after_trim):
+            raise error.TestFail("didn't get the bitmap of the target disk")
+        error.context("the bitmap_after_trim is %s" % bitmap_after_trim,
+                      logging.info)
+        total_block_after_trim = abs(sum([eval(i) for i in
+                                     bitmap_after_trim.split(',')]))
+        error.context("the total_block_after_trim is %d"
+                      % total_block_after_trim, logging.info)
+
+        if total_block_after_trim > total_block_before_trim:
+            raise error.TestFail("the bitmap_after_trim is lager, the command"
+                                 " guest-fstrim may not work")
+        if self.vm:
+            self.vm.destroy()
+
+    @error.context_aware
+    def gagent_check_get_interfaces(self, test, params, env):
+        """
+        Execute "guest-network-get-interfaces" command to guest agent
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        def find_interface_by_name(interface_list, target_interface):
+            """
+            find the specific network interface in the interface list return
+            by guest agent. return True if find successfully
+            """
+            for interface in interface_list:
+                if "target_interface" == interface["name"]:
+                    return True
+            error.context("didn't find the interface:%s in interface list"
+                          " return by guest agent" % target_interface)
```

Comment:
I think no need of these 2 lines, as there is error info when call this func(),
and logging.error is better than error.context.